include/yaracpp/types/yara_rule.h

```diff
@@ -41,21 +38,26 @@ class YaraRule
 
 		/// @name Getters
 		/// @{
-		YaraMeta* getMeta(std::string id);
+		YaraMeta* getMeta(const std::string &id);
 		YaraMatch* getMatch(std::size_t index);
 		YaraMatch* getFirstMatch();
 		/// @}
 
 		/// @name Setters
 		/// @{
-		void setName(std::string ruleName);
+		void setName(const std::string &ruleName);
 		/// @}
 
 		/// @name Other methods
 		/// @{
 		void addMeta(const YaraMeta &meta);
 		void addMatch(const YaraMatch &match);
 		/// @}
+
+        // @name Utility operator overloads
+        // @{
+        friend std::ostream& operator<<(std::ostream& o, const YaraRule& rule);
```

Comment:
I would add `#include <iosfwd>` into the header file for `std::ostream&` because otherwise, you might get `error: ‘ostream’ in namespace ‘std’ does not name a type`.

Also, I would add `#include <ostream>` into the `.cpp` file, as you call `std::ostream::operator<<()` in there, which requires a full definition of `std::ostream`.